frontend/src/components/GoalForm/GoalName.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import GoalNudge from './GoalNudge';
+import FormFieldThatIsSometimesReadOnly from './FormFieldThatIsSometimesReadOnly';
+import GoalText from './GoalText';
+
+export default function GoalName({
+  goalName,
+  goalNameError,
+  setGoalName,
+  validateGoalName,
+  isAppLoading,
+  recipient,
+  regionId,
+  selectedGrants,
+  onSelectNudgedGoal,
+  status,
+  isOnReport,
+  isNew,
+  userCanEdit,
+  isCurated,
+}) {
+  return (
+    <FormFieldThatIsSometimesReadOnly
```

Comment:
Note usage of new field.

Also: show the goal nudge options if creating a new goal. Do not show if the goal exists.